src/index.ts

```diff
@@ -1,23 +1,26 @@
 import { ReplaySubject } from 'rxjs/ReplaySubject'
 
-const subjects = new WeakMap()
+const subjects = new WeakMap();
 
 export function ObservableInput() {
-  return (target, propertyKey) => {
-    delete target[propertyKey]
+  return (target, prop) => {
```

Comment:
`prop` is ambiguous, suggests it could be the prop itself. `propName` is probably the better identifier.